graveler/committed/range_iterator.go

```diff
@@ -23,10 +23,12 @@ func (r *RangeIterator) Next() bool {
 		return false
 	}
 	val := r.it.Value()
+	// unmarshal value
 	var v *graveler.Value
 	if val.Value != nil {
 		v, r.err = UnmarshalValue(val.Value)
 		if r.err != nil {
+			r.value = nil
```

Comment:
Actually I don't particularly like this.  If `UnmarshalValue` decided to return a value, why erase it here?

Reply:
so .Err() will return the error and .Value() will return nil